### PR TITLE
Implemented on-screen keyboard theme support

### PR DESCRIPTION
**Pull request by kugiigi**
This implementation includes the following:
 - Support for loading the color value from JSON files
 - Replaced Border Images with Rectangle with borders (Magnifier, ExtendedKeysSelector, LanguageMenu)
 - Adding a new item in the schema (theme)
 - Updated LanguageMenu with the new ListItem component for better theme support
 - Added border support for each key

### CODE REVIEW

File: data/styles/ubuntu/themes/SuruBlack.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+	"fontColor": "#CDCDCD",
```
Comment (mariogrip) [inline data/styles/ubuntu/themes/SuruBlack.json]: Small issue, spacing here is wrong

File: qml/keys/ExtendedKeysSelector.qml [hunk 2]
```diff
@@ -64,11 +64,12 @@ KeyPopover {
...
         canvas.extendedKeysShown = enabled
     }
-
-    BorderImage {
-        id: popoverBackground
-
-        anchors.centerIn: anchorItem
+    
+    Rectangle{
+		id: popoverBackground
```
Comment (mariogrip) [inline qml/keys/ExtendedKeysSelector.qml]: spacing here too

File: qml/keys/ExtendedKeysSelector.qml [hunk 3]
```diff
@@ -79,8 +80,13 @@ KeyPopover {
         }
 
         height: rowOfKeys.height
-
-        source: "../images/popover.sci"
+        
+		color: fullScreenItem.theme.charKeyColor
```
Comment (mariogrip) [inline qml/keys/ExtendedKeysSelector.qml]: and here :)

File: qml/keys/Magnifier.qml [hunk 4]
```diff
@@ -46,11 +46,12 @@ KeyPopover {
...
         }
     }
-
-    BorderImage {
-        id: popper
-
-        width: parent.width
+    
+    Rectangle{
+		id: popper
```
Comment (mariogrip) [inline qml/keys/Magnifier.qml]: and here

File: qml/keys/Magnifier.qml [hunk 5]
```diff
@@ -60,8 +61,13 @@ KeyPopover {
         scale: animationStep
         transformOrigin: Item.Bottom
         opacity: animationStep
-
-        source: Qt.resolvedUrl("../images/magnified_key.sci")
+        
+		color: fullScreenItem.theme.charKeyColor
```
Comment (mariogrip) [inline qml/keys/Magnifier.qml]: ^

**Review by mariogrip — DISMISSED** [review]
other then some small nitpicking, it LGTM :+1

File: data/styles/ubuntu/themes/SuruBlack.json (hunk 1 above)
Comment (kugiigi) [reply data/styles/ubuntu/themes/SuruBlack.json]: fixed this, I had to search for the meaning of LGTM 😆

**Comment by cibersheep** [comment]
Let's test it in dev channel :)

**Review by UniversalSuperBox — APPROVED** [review]
I've done a cursory test of every language available manually on a few different themes, comparing them with the keyboard without these changes. I only found https://github.com/ubports/keyboard-component/issues/60, which happens with the keyboard without these changes.

The only possible problem I found is that the Japanese keyboard does not handle the theming very well. It's visible but doesn't honor your colors:

screenshot20181228_124356408

It's still readable and I'd be comfortable merging with this issue. It's up to you.

**Review by mariogrip — APPROVED** [review]
LGTM :+1:

**Comment by kugiigi** [comment]
maybe we can have separate bugs and PRs for layouts that has to be updated to support the themes.

**Comment by mariogrip** [comment]
@kugiigi Sounds like a good idea, let's merge this for now.

**Comment by 10JML01** [comment]
Device: BQ E5HD and M10FHD
Channel: rc
Version: 2018-W01

Works great!

**Comment by advocatux** [comment]
Device: Nexus 5 (hammerhead)
Channel: RC
Version: `2018-W01` (_really_ `2019-W01`)

Everything looks OK except the Japanese layout issue mentioned already.

**Comment by cibersheep** [comment]
Works great and even the tap & hold on space bar to move the cursor works better now

**Comment by erik-kueng** [comment]
Fairphone 2
rc
2018-W01

Works great.

**Comment by NeoTheThird** [comment]
Great job! Works very well.

**Comment by Stefano0101** [comment]
Fantastic job well done! Meizu Pro 5, testing Ota-7. 2018-W01

**Pull request merged**